L1Trigger/L1TGlobal/interface/L1TGlobalUtilHelper.h

```diff
@@ -131,8 +135,15 @@ namespace l1t {
     // Register the callback function with the Framework
     // if any InputTags still need to be found.
     if (findL1TAlgBlk || findL1TExtBlk) {
-      module.callWhenNewProductsRegistered([this, findL1TAlgBlk, findL1TExtBlk, iC](auto iBranch) {
-        checkToUpdateTags(iBranch, iC, findL1TAlgBlk, findL1TExtBlk);
+      auto const* pModule = &module;
```

Comment:
@Dr15Jones , thanks for this PR.

If I understand correctly, the best way forward would be to find all clients of `L1TGlobalUtilHelper` in CMSSW which initialise it incorrectly, fix them and then one could strip `L1TGlobalUtilHelper` of its current complexity (e.g. `existAs` calls, `checkToUpdateTags`, etc).

Would that be correct?

Reply:
@missirol I think you are correct that once all clients of `L1TGlobalUtilHelper` do proper initialization the class could be simplified.